src/functions-and-arrays.js

```diff
@@ -1,18 +1,80 @@
 // Iteration #1: Find the maximum
+var maxOfTwoNumbers = function(num1, num2){
+  if (num1 > num2){
+    return num1;
+  } else {
+    return num2;
+  }
+}
+
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+var findLongestWord = function(array){
+  if (array.length === 0){
+    return null
+  }
+  var longestWord = "";
+  for (i=0; i<=array.length-1; i++){
+    if (array[i].length > longestWord.length){
+      longestWord = array[i];
+        }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+var sumNumbers = function(array) {
+  if (array.length === 0) {
+    return 0;
+  }
+  var sum = 0;
+  for (i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
+
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+var averageNumbers = function(array) {
+  if (array.length === 0) {
+    return null;
+  }
+  var average = sumNumbers(array) / array.length;
+  return average;
+}
+
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+var averageWordLength = function(array){
+  if (array.length === 0) {
+    return null;
+  }
+  var totalLength = 0;
+  for (i = 0; i < array.length; i++) {
+    totalLength += array[i].length;
+  }
+  return totalLength / array.length;
+}
+
+//Bonus
+mixedArr = ["hi", 5, {fruit1: 'lemon', fruit2: 'orange'}, null];
+var avg =(array) =>{
+  if (array.length === 0) {
+    return null;
+  }
+  var totalLength = 0;
+  for (i = 0; i < array.length; i++) {
+    totalLength += array[i].length;
+  }
+}
```

Comment:
To pass all the Jasmine test, you could do something more easy maybe, like this:

let total = sum(array);
let average = total / array.length;

and then return that value, maybe toFixed( ).